lightning/src/chain/keysinterface.rs

```diff
@@ -1139,6 +1146,98 @@ impl KeysInterface for KeysManager {
 	}
 }
 
+/// Similar to [`KeysManager`], but allows the node using this struct to receive phantom node
+/// payments. `cross_node_seed` must be the same across all phantom-receiving nodes and also the
+/// same across restarts, or else payments may fail.
+pub struct PhantomKeysManager {
+	inner: KeysManager,
+	inbound_payment_key: KeyMaterial,
+	phantom_secret: SecretKey,
+}
+
+impl KeysInterface for PhantomKeysManager {
+	type Signer = InMemorySigner;
+
+	fn get_node_secret(&self) -> SecretKey {
+		self.inner.get_node_secret()
+	}
+
+	fn get_inbound_payment_key_material(&self) -> KeyMaterial {
+		self.inbound_payment_key.clone()
```

Comment:
It wasn't readily clear to me from the commit message why `inbound_payment_key` was needed instead of using the inner key. IIUC, it is because `ChannelManager` needs to authenticate the payment secret, which may have been created by one of many nodes. Is that accurate? However, the commit message speaks to only why `phantom_secret` is needed. Could you update the commit message? Perhaps the docs for `KeysInterface::get_inbound_payment_key_material` should mention authentication, too.

Reply:
Added more clarification to the commit message + the docs on `get_inbound_payment_key_material`.